### PR TITLE
Update DG: User stories

### LINKED ISSUE #36: Update DG: User stories
Week 7 Section 3 first point
 
- Target user profile
- Value proposition
- User stories

### PR DESCRIPTION
**Pull request by MAOXIONGKAI**
### **Update DG in the following ways:**
- update **target profile** and **value proposition** sections based on team project website
- update **user stories** section with the user stories from the project documentations on MS Team

### **Things to take note**:
Currently, some of the not feasible/questionable/duplicate user stories are not added to the DG. Some with poor phrasing/misleading descriptions are rewritten to make them more accurate.

Also, I have added some that may be useful/necessary in the DG as well.(under "Must have" or "Good to have" categories): 
- redo/undo actions
- add frequently accessed contacts as favorite(which allows them to be accessed using shorter commands)

Closes #36

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -274,27 +274,56 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* * *`  | Y1 CS student who wants to seek clarifications/actions from a professor regarding certain issues             | find contact details of the professor(s) for a course                                                                       | seek help/assistance with ease                                                      |
+| `* * *`  | Y1 CS student who is sick before a tutorial/lab/exam                                                         | find contact details of my TA                                                                                               | easily inform them about my absence on time                                         |
+| `* * *`  | Y1 CS student who is involved in any accident or emergency                                                   | find contact details of campus security                                                                                     | protect me and my friends' safety immediately                                       |
```
Comment (huangjiaxi1111) [inline docs/DeveloperGuide.md:308]: maybe finding public contact can be ** since we are focusing on maunally added contact for mvp

Comment (MAOXIONGKAI) [reply docs/DeveloperGuide.md]: I have changed them to ** now, can you help me confirm if the changes are all correct and help spot any other issues in the current PR, thanks!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -274,27 +274,56 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* * *`  | Y1 CS student who prefers a clean address book with useful contacts only                                     | delete contacts                                                                                                             | remove contacts that I no longer need to keep track of                               |
+| `* * *`  | Y1 CS student who hates using a mouse or touchpad                                                            | access to all functionalities of the app with keyboard                                                                      | manage the contacts in the most efficient way                                        |
+| `* * *`  | Y1 CS student who struggles to remember who to reach out to for certain issues                               | find the correct contact info by searching the category of the issue that I am trying to settle                             |                                                                                      |
```
Comment (huangjiaxi1111) [inline docs/DeveloperGuide.md:306]: I think here can be more general, something like:
Y1 CS student  |  search for contacts info from my address book |  contact them immediately
More specific use cases would be under **

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -274,27 +274,56 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* *`    | Y1 CS student who lives in campus and don't know where to contact with if I need to enquire some information | find contact details of campus hostels easily                                                                               |                                                                                      |
+| `* *`    | Y1 CS student who is keen to study together with friends/wants to manage friends know from class             | find out contact of friends who attend a specific module/course during the semester                                         |                                                                                      | 
+| `* *`    | Y1 CS student who wants to share my contact list with friends who need them                                  | transfer useful data to help others efficiently                                                                             |                                                                                      |
```
Comment (huangjiaxi1111) [inline docs/DeveloperGuide.md:319]: I think this one can be under * 
I'm not sure whether this is allowed

**Review by huangjiaxi1111 — APPROVED** [review]
👍

**Pull request merged**